metastore/src/test/scala/com/gu/tableversions/metastore/MetastoreSpec.scala

```diff
@@ -180,8 +182,8 @@ trait MetastoreSpec {
 
       } yield ()
 
-      val ex = the[Exception] thrownBy scenario.unsafeRunSync()
-      ex.getMessage.toLowerCase should include regex "partition not found"
+      val ex = the[Exception] thrownBy scenario.guarantee(tearDownUnderlyingTable).unsafeRunSync()
+      ex.getMessage.toLowerCase should include regex "notfound|not found"
```

Comment:
Any particular reason we want the error messages to be different for the two implementations?

Reply:
I just didn't bother to catch the aws exception and throw a custom one.}
Since this didn't check the specific error i assumed we didn't care as long as it said not found somewhere. The aws exception says entitynotfound or something like that without spaces
But I can catch it and throw an exception with a custom error message.

Reply:
I think it's nice to have well defined error messages that are the same irrespective of backend used. You can always include the underlying exception as a cause in the top level exception anyway.

Maybe consider doing something like this in `TableVersions`? https://github.com/guardian/table-versions/pull/11/files#diff-d0feac5d648547f94392ba148085a913R166

Reply:
I think it's good to have defined error messages that are the same regardless of implementation.
If we want to be able to handle the expected errors it would probably be better if instead of throwing exceptions we returned error in a more FP way like in an Either or a custom sealed trait.

If we stick to throwing exceptions it would be more useful if we definen different exception classes that can be caught and handled independently

Reply:
We're not really _throwing_ exceptions though, we're just using `Throwable` objects as part of failed `IO`s as that's what it uses. But I guess one option is to use some monad stack e.g. `EitherT[IO, E, T]` or something. But that's a bigger discussion of course...

As for this case, it's not a big deal, but it seems odd if the tests for the base trait need to know all the variations of error messages that the implementations may produce. I guess another is to not check the returned error at all.

Reply:
yeah I agree that the test should not have to handle different cases like this.
maybe for now we can check that some error was returned and also an additional check we could make is that the version has not changed